ProcessLib/HeatTransportBHE/CreateHeatTransportBHEProcess.cpp

```diff
@@ -184,23 +190,82 @@ std::unique_ptr<Process> createHeatTransportBHEProcess(
         if (bhe_type == "1U")
         {
             bhes.push_back(BHE::createBHE1U(bhe_config, curves));
+            // find if bhe uses python boundary condition
+            BHE::BHE_1U individual_bhe = boost::get<BHE::BHE_1U>(bhes[0]);
+            if (individual_bhe.bhe_if_use_python_bc == true)
+            {
+                if_bhe_network_exist_python_bc = true;
+            }
             continue;
         }
+
         OGS_FATAL("Unknown BHE type '%s'.", bhe_type.c_str());
     }
     // end of reading BHE parameters -------------------------------------------
 
-    HeatTransportBHEProcessData process_data{thermal_conductivity_solid,
-                                             thermal_conductivity_fluid,
-                                             thermal_conductivity_gas,
-                                             heat_capacity_solid,
-                                             heat_capacity_fluid,
-                                             heat_capacity_gas,
-                                             density_solid,
-                                             density_fluid,
-                                             density_gas,
-                                             std::move(bhes)};
+    //! Python object computing BC values.
+    HeatTransportBHEProcessData* process_data;
 
+    if (if_bhe_network_exist_python_bc == false)
+    {
+        process_data =
+            new HeatTransportBHEProcessData(thermal_conductivity_solid,
```

Comment:
`new` and `delete` are nowadays replaced by `unique_ptr` (or other smart pointers). In the particular case I don't see why a pointer is needed: there is no base class used here.

Reply:
fixed in commit #32114c54